File: examples/Delphi/ExaBtn2.pas
unit Exabtn2;

interface

uses
  SysUtils, WinTypes, WinProcs, Messages, Classes, Graphics, Controls,
  Forms, Dialogs, ExtCtrls, OvcBase, OvcCal, StdCtrls, Buttons, OvcEF,
  OvcData, OvcSF, Menus, OvcSc;

type
  TfrmCal = class(TForm)
    Calendar: TOvcCalendar;
    OvcController1: TOvcController;
    Panel1: TPanel;
    BitBtn1: TBitBtn;
    BitBtn2: TBitBtn;
    procedure CalendarDblClick(Sender: TObject);
    procedure FormKeyDown(Sender: TObject; var Key: Word;
      Shift: TShiftState);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmCal: TfrmCal;

implementation

{$R *.DFM}

procedure TfrmCal.CalendarDblClick(Sender: TObject);
begin
  ModalResult := mrOK;
end;

procedure TfrmCal.FormKeyDown(Sender: TObject; var Key: Word;
  Shift: TShiftState);
begin
  case Key of
    VK_F1,                                                             {!!.01}
    VK_RETURN,
    VK_ESCAPE,
    VK_NONE,
    VK_SHIFT,
    VK_CONTROL,
    VK_ALT,
    VK_CAPITAL,
    VK_NUMLOCK,
    VK_SCROLL,
    VK_UP,
    VK_DOWN,
    VK_LEFT,
    VK_RIGHT,
    VK_HOME,
    VK_END,
    VK_PRIOR,
    VK_NEXT,
    VK_SPACE,
    VK_TAB : {};
  else
    ModalResult := mrCancel;
  end;
end;

end.

File: examples/CBuildr3/ExABtn2.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "ExABtn2.h"
//---------------------------------------------------------------------------
#pragma link "OvcCal"
#pragma link "OvcBase"
#pragma link "OvcSC"
#pragma resource "*.dfm"
TfrmCal *frmCal;
//---------------------------------------------------------------------------
__fastcall TfrmCal::TfrmCal(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmCal::CalendarDblClick(TObject *Sender)
{
  ModalResult = mrOk;
}
//---------------------------------------------------------------------------
void __fastcall TfrmCal::Spinner1Click(TObject *Sender, TOvcSpinState State,
  double Delta, bool Wrap)
{
  switch (State) {
    case ssLeftBtn  : Calendar->IncMonth(-Delta); break;
    case ssRightBtn : Calendar->IncMonth(+Delta); break;
  }
}
//---------------------------------------------------------------------------
void __fastcall TfrmCal::Spinner2Click(TObject *Sender, TOvcSpinState State,
  double Delta, bool Wrap)
{
  switch (State) {
    case ssLeftBtn  : Calendar->IncYear(-Delta); break;
    case ssRightBtn : Calendar->IncYear(+Delta); break;
  }
}
//---------------------------------------------------------------------------
void __fastcall TfrmCal::FormKeyDown(TObject *Sender, WORD &Key,
  TShiftState Shift)
{
  switch (Key) {
    case VK_F1   :
    case VK_RETURN :
    case VK_ESCAPE :
    case VK_NONE :
    case VK_SHIFT :
    case VK_CONTROL :
    case VK_ALT :
    case VK_CAPITAL :
    case VK_NUMLOCK :
    case VK_SCROLL :
    case VK_UP :
    case VK_DOWN :
    case VK_LEFT :
    case VK_RIGHT :
    case VK_HOME :
    case VK_END :
    case VK_PRIOR :
    case VK_NEXT :
    case VK_SPACE :
    case VK_TAB : break;
    default : ModalResult = mrCancel;
  }
}
//---------------------------------------------------------------------------
void __fastcall TfrmCal::FormCreate(TObject *Sender)
{
  BitBtn1->Caption = "";
  BitBtn2->Caption = "";  
}
//---------------------------------------------------------------------------
